klippa/src/name.rs

```diff
@@ -0,0 +1,123 @@
+    let name_records = name.name_record();
+    //TODO: support name_table_override
+    //TODO: support name table version 1
+    let mut retained_name_record_idxes = name_records
+        .iter()
+        .enumerate()
+        .filter(|(_, record)| {
+            if !plan.name_ids.contains(record.name_id())
+                || !plan.name_languages.contains(record.language_id())
+            {
+                return false;
+            }
+            plan.subset_flags
+                .contains(SubsetFlags::SUBSET_FLAGS_NAME_LEGACY)
+                || record.is_unicode()
+        })
+        .map(|x| x.0)
+        .collect::<Vec<_>>();
+
+    retained_name_record_idxes.sort_unstable_by(|a, b| {
+        let id_a = name_records[*a].platform_id();
+        let id_b = name_records[*b].platform_id();
+        if id_a != id_b {
+            return id_a.cmp(&id_b);
+        }
+
+        let id_a = name_records[*a].encoding_id();
+        let id_b = name_records[*b].encoding_id();
+        if id_a != id_b {
+            return id_a.cmp(&id_b);
+        }
+
+        let id_a = name_records[*a].language_id();
+        let id_b = name_records[*b].language_id();
+        if id_a != id_b {
+            return id_a.cmp(&id_b);
+        }
+
+        let id_a = name_records[*a].name_id().to_u16();
+        let id_b = name_records[*b].name_id().to_u16();
+        if id_a != id_b {
+            return id_a.cmp(&id_b);
+        }
+
+        let len_a = name_records[*a].length();
+        let len_b = name_records[*b].length();
+        if len_a != len_b {
+            return len_a.cmp(&len_b);
+        }
+
+        Ordering::Equal
+    });
+
+    let name_data = name.offset_data().as_bytes();
+    let mut out = Vec::with_capacity(name_data.len());
+    // version
+    // TODO: support version 1
+    out.extend_from_slice(&[0, 0]);
+    //count
+    let count = retained_name_record_idxes.len() as u16;
+    out.extend_from_slice(&count.to_be_bytes());
+    //storage_offset
+    let storage_offset = count * 12 + 6;
+    out.extend_from_slice(&storage_offset.to_be_bytes());
+
+    //pre-allocate space for name records array
+    out.resize(storage_offset as usize, 0);
+
+    let mut string_offset = 0_u16;
+    let storage_start = name.storage_offset() as usize;
+    for (new_idx, old_idx) in retained_name_record_idxes.iter().enumerate() {
+        let old_record_start = 6 + 12 * old_idx;
+        let new_record_start = 6 + 12 * new_idx;
+        //copy name_record except for string offset
+        out.get_mut(new_record_start..new_record_start + 10)
+            .unwrap()
+            .copy_from_slice(
+                name_data
+                    .get(old_record_start..old_record_start + 10)
+                    .unwrap(),
+            );
+        //copy string offset
+        out.get_mut(new_record_start + 10..new_record_start + 12)
+            .unwrap()
+            .copy_from_slice(&string_offset.to_be_bytes());
```

Comment:
idx * 12 + 6 repeats several simes and may warrant a helper function.

the literals 6, 10, and 12 feel like they want _NAMED_CONSTANTS.